### ISSUE TITLE
Training the context network

### ISSUE THREAD

**Issue by saxenarohan97** [NVlabs/PWC-Net#27]
Hi,
Your paper has very interesting ideas and I thoroughly enjoyed reading it. I had a couple of questions about the paper:

### Question 1
I had got my hands on an earlier version of the paper. In this version, it was mentioned that:
> We jointly learn all the parameters, including those of the context network.

However, in the same paper it was said that the loss function used while training is:
![image](https://user-images.githubusercontent.com/17175695/44217561-09f50480-a178-11e8-90f2-d6262cfc005e.png)

The `w(x)` used in the above equation has been used to denote the output of the optical flow estimator network. If this loss function is used, I don't see how the context network would be trained. Should the equation use `ŵ(x)` instead of `w(x)`, where `ŵ(x)` refers to the output of the context network? This will ensure that the full network is indeed trained end-to-end.

### Question 2
This line about learning all the parameters together has been removed in a recent version of the paper. Is that because you finally trained the context network separately from the rest of the pipeline? If that is the case, what loss function did you use when training just the context network?

**Comment by luvegood** [comment]
I believe the full network is trained end-to-end, in the meantime, you may refer to the training protocol in caffe to see the actual training pipeline: https://github.com/NVlabs/PWC-Net/blob/master/Caffe/model/train.prototxt

**Comment by deqings** [comment]
Question 1: yes, you are correct, we should use ŵ(x) at the last pyramid level that uses the context network. We will clarify this in an updated version. Thanks for pointing out the issue.

Question 2: all the parameters are trained jointly. Sorry for the confusion.
